src/subscribables/dependentObservable.js

```diff
@@ -1,66 +1,175 @@
-        isSleeping = false;
-
-    if (readFunction && typeof readFunction == "object") {
+    if (typeof evaluatorFunctionOrOptions === "object") {
         // Single-parameter syntax - everything is on this "options" param
-        options = readFunction;
-        readFunction = options["read"];
+        options = evaluatorFunctionOrOptions;
     } else {
         // Multi-parameter syntax - construct the options according to the params passed
         options = options || {};
-        if (!readFunction)
-            readFunction = options["read"];
+        if (evaluatorFunctionOrOptions) {
+            options["read"] = evaluatorFunctionOrOptions;
+        }
     }
-    if (typeof readFunction != "function")
+    if (typeof options["read"] != "function")
         throw new Error("Pass a function that returns the value of the ko.computed");
 
-    function addDependencyTracking(id, target, trackingObj) {
-        if (pure && target === dependentObservable) {
-            throw Error("A 'pure' computed must not be called recursively");
+    var state = {
+        _latestValue: undefined,
+        _needsEvaluation: true,
+        _isBeingEvaluated: false,
+        _suppressDisposalUntilDisposeWhenReturnsFalse: false,
+        _isDisposed: false,
+        pure: false,
+        isSleeping: false,
+        readFunction: options["read"],
+        writeFunction: options["write"],
+        disposeWhenNodeIsRemoved: options["disposeWhenNodeIsRemoved"] || options.disposeWhenNodeIsRemoved || null,
+        disposeWhenOption: options["disposeWhen"] || options.disposeWhen,
+        disposeWhen: options["disposeWhen"] || options.disposeWhen,
+        dependencyTracking: {},
+        _dependenciesCount: 0,
+        evaluationTimeoutInstance: null,
+        evaluatorFunctionTarget: evaluatorFunctionTarget || options["owner"],
+        options: options
+    };
+
+    function dependentObservable() {
+        if (arguments.length > 0) {
+            if (typeof state.writeFunction === "function") {
+                // Writing a value
+                state.writeFunction.apply(state.evaluatorFunctionTarget, arguments);
+            } else {
+                throw new Error("Cannot write a value to a ko.computed unless you specify a 'write' option. If you wish to read the current value, don't pass any parameters.");
+            }
+            return this; // Permits chained assignments
+        } else {
+            // Reading the value
+            ko.dependencyDetection.registerDependency(dependentObservable);
+            if (state._needsEvaluation || (state.isSleeping && dependentObservable.haveDependenciesChanged())) {
+                dependentObservable.evaluateImmediate();
+            }
+            return state._latestValue;
         }
+    }
 
-        dependencyTracking[id] = trackingObj;
-        trackingObj._order = _dependenciesCount++;
-        trackingObj._version = target.getVersion();
+    dependentObservable.state = state;
```

Comment:
Good point. It will be private-named by the minifier anyway, but I agree it would be better still to use a `Symbol`, equivalent to how we do for `ko.observable`.